src/main.ts

```diff
@@ -48,6 +48,14 @@ async function run(): Promise<void> {
     if (!('GORELEASER_CURRENT_TAG' in process.env)) {
       process.env.GORELEASER_CURRENT_TAG = tag;
     }
+
+    if (isTagDirty) {
+      core.setOutput("tag", `${tag}-next`);
+    } else {
+      core.setOutput("tag", tag);
+    }
```

Comment:
I don't think we should have `-next` hardcoded like this. It should reflect the `.goreleaser.yml` provided.

I wonder if somehow GoReleaser should produce a metadata file (e.g., `metadata.json`) that contains the list of [known/generated templates vars](https://goreleaser.com/customization/templates/#name-templates) like `.Version`, etc... so we could use it as a `metadata` output in GitHub Actions:

```yaml
      -
        name: Run GoReleaser
        id: goreleaser
        uses: goreleaser/goreleaser-action@v2
        with:
          version: latest
          args: release --rm-dist
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      -
        run: |
          echo "${{ fromJSON(steps.goreleaser.outputs.metadata).version }}"
```

Thoughts @caarlos0?

Reply:
That would work as well. Just looking to deploy a fluent version.

Reply:
@crazy-max Btw can you point out how `.goreleaser.yml` can influence the name? Wasn't aware of that.

Reply:
I thinks it's a good idea @crazy-max , we already do it for artifacts...

Reply:
@till the snapshot name template config 


And also the nightly on goreleaser pro

Reply:
@caarlos0 Would you be open if I adjust that then? I don't want to parse another file in my workflow. An output would be preferred. :)

Reply:
yeah, we need to add this feature to goreleaser first (e.g. a `dist/context.json`) or something like that, the IMHO we can parse it here and output the values...

things I think would be useful as outputs:

- current and previous tag names and versions
- project name
- what else?

artifacts can be get from `dist/artifacts.json`

Reply:
Maybe isDirty, or isSnapshot? Can't think of much right now.

Reply:
yeah, sound good, will open an issue on goreleaser

Reply:
https://github.com/goreleaser/goreleaser/issues/2835

Reply:
> and output the values...

Yeah we just need one output named `metadata` or `context` that will be the JSON representation so we can use `fromJSON()` in the underlying steps.

We do this in the [metadata-action](https://github.com/docker/metadata-action#json-output-object):

https://github.com/docker/metadata-action/blob/2cad85de2462ab92a22ff973a347fb25c91d6082/src/main.ts#L62-L67

![image](https://user-images.githubusercontent.com/1951866/149614567-5c5cac6d-0384-47f4-99f0-8ff7d23061d6.png)

> what else?

The more the better :pray: